rinex-cli/src/main.rs

```diff
@@ -74,77 +69,97 @@ pub fn create_workspace(path: PathBuf) {
 }
 
 /*
- * Creates File/Data context defined by user.
- * Regroups all provided files/folders,
+ * Macro to build QcExtraData of a specific RINEX Type
  */
-fn create_context(cli: &Cli) -> QcContext {
-    QcContext {
-        primary: QcInputData::new(cli.input_path()).expect("failed to parse primary RINEX file"),
-        nav: {
-            if let Some(path) = cli.nav_path() {
-                if let Ok(ctx) = QcInputData::new(path) {
-                    if ctx.rinex.is_navigation_rinex() {
-                        trace!("nav augmentation \"{}\"", path);
-                        Some(ctx)
-                    } else {
-                        error!("--nav must be valid navigation data");
-                        None
-                    }
-                } else {
-                    error!("failed to parse nav augmentation \"{}\"", path);
-                    None
+fn build_extra_rinex_data(
+    paths: Option<ValuesRef<'_, String>>,
+    rtype: RinexType,
+) -> Option<QcExtraData<Rinex>> {
+    if let Some(paths) = paths {
+        let mut ctx = QcExtraData::<Rinex>::default();
+        for path in paths {
+            if let Ok(new) = Rinex::from_file(path) {
+                if new.header.rinex_type != rtype {
+                    let stem = Path::new(path).file_stem().unwrap();
+                    error!("\"{}\" : invalid {} RINEX", stem.to_string_lossy(), rtype);
+                    continue;
                 }
-            } else {
-                None
-            }
-        },
-        atx: {
-            if let Some(path) = cli.atx_path() {
-                if let Ok(ctx) = QcInputData::new(path) {
-                    if ctx.rinex.is_antex() {
-                        trace!("atx data \"{}\"", path);
-                        Some(ctx)
-                    } else {
-                        error!("--atx should be valid antex data");
-                        None
-                    }
+                if ctx.paths().len() == 0 {
+                    // first file passed by user
+                    ctx = QcExtraData {
+                        paths: vec![Path::new(path).to_path_buf()],
+                        data: new.clone(),
+                    };
+                    trace!("{} file \"{}\"", rtype, path);
                 } else {
-                    error!("failed to parse atx data \"{}\"", path);
-                    None
+                    match ctx.data_mut().merge_mut(&new) {
+                        Ok(_) => {
+                            trace!("{} file \"{}\"", rtype, path);
+                            ctx.paths.push(Path::new(path).to_path_buf());
```

Comment:
`PathBuf::from`

Reply:
`PathBuf::from` can't apply here, because `path` does not have a known size at compilation time ([u8]). Path comes from 

```rust
for path in clap::ValuesRef<'_, String> 
```

and clap::ValuesRef is the only way I know, at the moment, to have a command line argument pass more than one value

Reply:
the cli.rs structure should cast to Strings and not simply return ValuesRef then